app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+class User < ActiveRecord::Base
+  has_many :products
+  has_many :orders
+  has_many :reviews
+
+  validates :name, presence: true
+  validates :user_name, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
```

Comment:
This totally works, especially since you are not allowed to use the devise gem for this project.

In the long term you will want to validate emails by sending the email and getting a confirmation because it is really hard to debug regular expressions.